spccore/internal/lock.py

```diff
@@ -81,14 +94,41 @@ def release(self) -> None:
 
         :raises OSError: when it fails to release a lock
         """
-        if self.held:
+        if self._has_lock():
```

Comment:
Thanks @zimingd. This sounds like a test case. I will add a test case to verify this, and make the recommended changes.

Reply:
I added a test case as described above. It doesn't fail. But it could mean that my computer is too slow to producing the bug. I kept both the test case and the suggested change.